src/main/java/uk/gov/companieshouse/authcodenotification/service/PrivateDataRetrievalService.java

```diff
@@ -40,4 +45,25 @@ public OverseasEntityDataApi getOverseasEntityData(String requestId, String comp
             throw new ServiceException(e.getMessage(), e);
         }
     }
+
+    public CompanyProfileApi getCompanyProfile(String requestId, String companyNumber)
```

Comment:
We should use the non-private sdk to get the company profile, the private one looks like it was only added last year so i don't think it is the one we were expecting. So we'll need a PublicDataRetrievalService and in ApiClientService we'll need a getApiClient using ApiSdkManager.getSdk()

Reply:
Why is it not the one we were expecting? We seem to have the option of getting the company profile from both a public and private sdk and both are being used currently in the CH codebase in different places. Do we know what criteria makes the public one correct and the private one wrong and vice-versa?

Reply:
Yes, agree. I think the criteria are: if we're getting public data we should use the public SDK and if we're getting private data, the private SDK. Because company profile is public data, it may end up being removed from the private SDK and we want to avoid having to re-work this service at that point.

Reply:
Using the public api call.